### ISSUE TITLE
unable to create a sweep with empty parameters dict

### ISSUE THREAD

**Issue by evfro** [wandb/client#2977]
**Describe the bug**
Unable to create a sweep via Web UI with the following config:
```yaml
method: random
name: test
parameters: {}
```

**To Reproduce**
1. Go to a project sweeps
2. Click on `Create sweep` button
3. Paste config from above into the sweep window
4. Click on `Initialize Sweep` button
4. See error "Invalid sweep config: Parameters section of sweep config must be a dict of at least length 1"

**Expected behavior**
In previous versions I could create sweeps with empty parameters dict.

**Screenshots**
![image](https://user-images.githubusercontent.com/5283394/144477692-0e4eccd5-20e5-4df0-8f38-2c361f92bf07.png)

**Comment by dannygoldstein** [comment]
Hi Evgeny,

Out of curiosity, why do you want to create a sweep with no parameters?

**Comment by evfro** [comment]
Hi Danny,

I'm testing a non-trivial grid-search with additional internal dependencies, which cannot be configured at the wandb side.

So, I created my own grid-search routine that generates grid points from the local code and I only use wandb for logging results, not for suggesting grid points. Hence, 1) I need to spin agents for an indefinite number of times , and 2) I don't need any configuration suggestions. That's why 1) I set `method: random` and 2) `parameters: { }` (they get populated on the fly). Hope, explanation is more or less clear.
